packages/core/src/stateUtils.ts

```diff
@@ -1153,10 +1164,8 @@ function enterStates(
     actions.push(...stateNodeToEnter.entry);
 
     if (statesForDefaultEntry.has(stateNodeToEnter)) {
-      for (const stateNode of statesForDefaultEntry) {
-        const initialActions = stateNode.initial!.actions;
-        actions.push(...initialActions);
-      }
```

Comment:
this was quite wrong, I was surprised that the test suite passed with it - I have a strong feeling that we might be missing some test case, I'll try to figure it out tomorrow